### PR TITLE
SHOP-11: Admin user on startup, adding products to database

### PR DESCRIPTION
**Pull request by Wraan**
Added Admin User on startup, added Add Product option under /admin/addProduct mapping with validation, changed ProductSpecification entity to show orderNumber

### CODE REVIEW

File: src/main/java/com/shop/dto/ProductDto.java [hunk 1]
```diff
@@ -15,8 +17,10 @@
     @NotNull
     private int amount;
     @NotBlank
+    @Length(max = 50)
     private String name;
     @NotBlank
+    @Length(max = 20)
```
Comment (Swistak06) [inline src/main/java/com/shop/dto/ProductDto.java]: Length for producer should be at least set at 50.

**Review by Swistak06 — COMMENTED** [review]
Generally it is ok.

File: src/main/java/com/shop/controller/ProductController.java [hunk 2]
```diff
@@ -1,34 +1,48 @@
...
     ProductService productService;
 
-    @GetMapping("/product")
-    public String showProductPage() {
+    @GetMapping("/product/{id}")
+    public String showProductPage(@PathVariable("id") long id, Model model) {
+        Product product = productService.findById(id);
+        if(product == null)
+            return "redirect:/";
+        model.addAttribute("product", product);
```
Comment (polskikiel) [inline src/main/java/com/shop/controller/ProductController.java]: add space after return or add {}

File: src/main/java/com/shop/service/DtoServiceImpl.java [hunk 3]
```diff
@@ -26,4 +30,29 @@ private Address createAddressFromAddressDto(User user, AddressDto addressDto){
...
     }
+
+    public List<SpecificationDto> getListSpecificationDtoFromJson(String json){
+        List<SpecificationDto> specificationDtoList;
+        try {
+            SpecificationDto[] specArray = new ObjectMapper().readValue(json, SpecificationDto[].class);
+            specificationDtoList = Arrays.asList(specArray);
+        }
+        catch(IOException e){
+            specificationDtoList = null;
```
Comment (polskikiel) [inline src/main/java/com/shop/service/DtoServiceImpl.java]: unnecessary

File: src/main/java/com/shop/service/ProductServiceImpl.java [hunk 4]
```diff
@@ -1,17 +1,31 @@
...
 @Service
 public class ProductServiceImpl implements ProductService {
 
     @Autowired
     ProductRepository productRepository;
 
+    @Autowired
+    DtoService dtoService;
+
+    @Autowired
```
Comment (polskikiel) [inline src/main/java/com/shop/service/ProductServiceImpl.java]: isn't better to inject by constructor here>

File: src/main/java/com/shop/model/ProductSpecification.java [hunk 5]
```diff
@@ -18,13 +18,23 @@
     @ManyToOne(cascade = CascadeType.ALL)
     @JoinColumn(name = "specification_id")
     private Specification specification;
+    private int orderNumber;
```
Comment (polskikiel) [inline src/main/java/com/shop/model/ProductSpecification.java:21]: why do you introduce that field?

Comment (Wraan) [reply src/main/java/com/shop/model/ProductSpecification.java:21]: to show specifications in a specific order

File: src/main/java/com/shop/service/ProductServiceImpl.java (hunk 4 above)
Comment (Wraan) [reply src/main/java/com/shop/service/ProductServiceImpl.java]: sometimes injecting by constructor may make a loop, will check it

File: src/main/java/com/shop/service/DtoServiceImpl.java (hunk 3 above)
Comment (Wraan) [reply src/main/java/com/shop/service/DtoServiceImpl.java]: done

File: src/main/java/com/shop/controller/ProductController.java (hunk 2 above)
Comment (Wraan) [reply src/main/java/com/shop/controller/ProductController.java]: done

**Review by Swistak06 — APPROVED** [review]
Corrections were done. I approve this pull request.

**Pull request merged**